src/bsp/src/stm32/include/Decawave.h

```diff
@@ -125,6 +139,81 @@ class Decawave {
                                    uint32_t rxAfterTxTimeUs,
                                    UWBRxFrame& frame,
                                    uint16_t rxTimeoutUs);
+    /**
+     * @brief Transmits data over UWB and blocks until response is received
+     * @param buf Buffer to transmit
+     * @param length Length of the data
+     * @param rxStartDelayUS Number of microseconds between the TX and the start of the RX
+     * @param frame [out] Variable in which all frame data will be put once packet is received
+     * @param rxTimeoutUs Timeout in microseconds. 0 to block indefinitely
+     * @return True if successful, false otherwise.
+     */
+    bool transmitAndReceiveDelayed(uint8_t* buf,
+                                   uint16_t length,
+                                   uint32_t rxStartDelayUS,
+                                   UWBRxFrame& frame,
+                                   uint16_t rxTimeoutUs);
+
+    /**
+     * @brief Sets transmission antenna delay in DecawaveTimeUnits
+     * @param delay Delay in DecawaveTimeUnits
+     */
+    void setTxAntennaDLY(uint16 delay);
+
+    /**
+     * @brief Sets reception antenna delay in DecawaveTimeUnits
+     * @param delay Delay in DecawaveTimeUnits
+     */
+    void setRxAntennaDLY(uint16 delay);
+
+    /**
+     * @brief Get transmission antenna delay in DecawaveTimeUnits
+     * @return Delay in DecawaveTimeUnits
+     */
+    uint16_t getTxAntennaDLY();
+
+    /**
+     * @brief Get reception antenna delay in DecawaveTimeUnits
+     * @return Delay in DecawaveTimeUnits
+     */
+    uint16_t getRxAntennaDLY();
+
+    /**
+     * @brief Retrieves transmission timestamp in DecawaveTimeUnits
+     * @param txTimestamp Transmission timestamp in DecawaveTimeUnits
+     */
+    void getTxTimestamp(uint64_t* txTimestamp);
+
+    /**
+     * @brief Retrieves reception timestamp in DecawaveTimeUnits
+     * @param rxTimestamp Reception timestamp in DecawaveTimeUnits
+     */
+    void getRxTimestamp(uint64_t* rxTimestamp);
+
+    /**
+     * @brief Retrieves present time DecawaveTimeUnits
+     * @param sysTime Present timestamp in DecawaveTimeUnits
+     */
+    void getSysTime(uint64_t* sysTime);
+
+    /**
+     * @brief Retrieves the present state of calibration
+     * @return The present state of calibration
+     */
+    DW_STATE getState();
+
+    /**
+     * @brief Sets the next state of calibration
+     * @param state The next state of calibration
+     */
+    void setState(DW_STATE state);
+
+    /**
+     * @brief Sets a 5 byte timestamp into the specied message field
+     * @param ts Timestamp to insert in message
+     * @param tsField Field to put timestamp in
+     */
+    void finalMsgAddTs(uint8* tsField, uint64_t ts);
```

Comment:
Not sure this belongs in the decawave class. It is specific for a single message.